libqtile/widget/memory.py

```diff
@@ -39,27 +39,33 @@ class Memory(base.ThreadedPollText):
     SwapFree: Returns amount of swap free
     SwapUsed: Returns amount of swap in use
 """
+
     orientations = base.ORIENTATION_HORIZONTAL
     defaults = [
-        ("fmt", "{MemUsed}M/{MemTotal}M", "Formatting for field names.")
+        ("format", "{MemUsed}M/{MemTotal}M", "Formatting for field names."),
+        ("update_interval", 1.0, "Update interval for the Memory"),
     ]
 
     def __init__(self, **config):
         super().__init__(**config)
         self.add_defaults(Memory.defaults)
 
+    def tick(self):
+        self.update(self.poll())
+        return self.update_interval
```

Comment:
What's the rationale for adding this? This seems to be changing the widget to behave like the in-loop poll text base class.

Reply:
If I remember correctly, without that it updated really randomly, occasionally it took a bit over second, sometimes over 3 seconds. With that it takes more consistently a bit over 1 second to update.

Reply:
I think the reason that is the case is that it will run the update in the main thread, if that is desired it should base class the in-loop poll text widget, rather than subclassing the threaded poll text widget, but only if we know the function calls in the poll function are fast and not going to block.